proposals/generate_validating_admission_policies.md

```diff
@@ -468,9 +473,171 @@ spec:
+
+  Another solution is to exclude resources via the ValidatingAdmissionPolicyBinding as follows:
+
+  ```yaml
+  apiVersion: admissionregistration.k8s.io/v1beta1
+  kind: ValidatingAdmissionPolicyBinding
+  metadata:
+    name: "check-deployment-labels-binding"
+  spec:
+    policyName: "check-deployment-labels"
+    validationActions: [Deny]
+    matchResources:
+      excludeResourceRules:
+      - apiGroups:     ["apps"]
+        apiVersions:   ["v1"]
+        operations:    ["CREATE", "UPDATE"]
+        resources:     ["deployments"]
+        resourceNames: ["important-tool"]
+      namespaceSelector:
+        matchLabels:
+          environment: staging
+  ```
+
+  However, `matchResources` in the binding is intersected with the VAP's `matchConstraints`, so only requests that are matched by the VAP can be selected by this. It means that the policy will match/exclude deployments in a namespace whose `environment` label is set to `staging`. Therefore, excluding resources in the VAP binding has the same issues as the VAP.
+
+- Given an exception that excludes Deployments named `important-tool` in the `delta` namespace.
+
+  ```yaml
+  apiVersion: kyverno.io/v2beta1
+  kind: PolicyException
+  metadata:
+    name: delta-exception
+    namespace: delta
+  spec:
+    exceptions:
+    - policyName: disallow-host-namespaces
+      ruleNames:
+      - host-namespaces
+    match:
+      any:
+      - resources:
+          kinds:
+          - Deployment
+          namespaces:
+          - delta
+          names:
+          - important-tool
+  ```
+  
+  Both VAPs and their binding can't specify namespaces by their names. Instead, they use the `namespaceSelector` to match resources whose namespace has a specific label. Therefore, the label `kubernetes.io/metadata.name: <namespace-name>` can be used in either VAP or its binding.
+
+  ```yaml
+  apiVersion: admissionregistration.k8s.io/v1beta1
+  kind: ValidatingAdmissionPolicy
+  metadata:
+    name: "check-deployment-labels"
+  spec:
+    failurePolicy: Fail
+    matchConstraints:
+      excludeResourceRules:
+      - apiGroups:     ["apps"]
+        apiVersions:   ["v1"]
+        operations:    ["CREATE", "UPDATE"]
+        resources:     ["deployments"]
+        resourceNames: ["important-tool"]
+      namespaceSelector:
+        matchLabels:
+          kubernetes.io/metadata.name: delta
+  ```
+
+  However, the use of `namespaceSelector` limits resource matching to the same namespace as the exclusion.
+
+- An exception that matches resources based on annotations, subjects, roles, or clusterRoles cannot be converted to VAPs.
+
+In conclusion, exceptions provide the flexibility to exclude resources that VAPs lack, making it impossible to generate a VAP from an exception. As a result, if an exception is created for a Kyverno policy that uses `validate.cel` subrule, the Kyverno engine will handle the resource validation itself instead of generating the corresponding VAPs.
```

Comment:
Can we allow use of some subset of exceptions?

Reply:
For the following cases, I think a VAP can be generated:
1. Matching all resources and excluding a specific one.
2. Matching and excluding in the same namespace.

However, there are a lot of limitations and I don't know if it is worth considering this feature.

Reply:
Can we clarify the limitations and propose changes to ValidatingAdmissionPolicy resources?

https://github.com/kubernetes/kubernetes/issues/122205#issuecomment-1927390823